src/client.ts

```diff
@@ -2922,12 +2924,27 @@ export class StreamChat<StreamChatGenerics extends ExtendableGenerics = DefaultG
     if (this.userAgent) {
       return this.userAgent;
     }
+
     const version = process.env.PKG_VERSION;
+    const clientBundle = process.env.CLIENT_BUNDLE;
+
+    let userAgentString = '';
     if (this.sdkIdentifier) {
-      return `stream-chat-${this.sdkIdentifier.name}-v${this.sdkIdentifier.version}-llc-v${version}`;
+      userAgentString = `stream-chat-${this.sdkIdentifier.name}-v${this.sdkIdentifier.version}-llc-v${version}`;
     } else {
-      return `stream-chat-js-v${version}-${this.node ? 'node' : 'browser'}`;
+      userAgentString = `stream-chat-js-v${version}-${this.node ? 'node' : 'browser'}`;
     }
+
+    const { os, model } = this.deviceIdentifier ?? {};
+
+    return ([
+      // reports the device OS, if provided
+      ['os', os],
+      // reports the device model, if provided
+      ['device_model', model],
+      // reports which bundle is being picked from the exports
+      ['client_bundle', clientBundle],
+    ] as const).reduce((acc, [key, value]) => (value ? acc.concat(`|${key}=${value ?? ''}`) : acc), userAgentString);
```

Comment:
```suggestion
    ] as const).reduce((acc, [key, value]) => (value ? acc.concat(`|${key}=${value}`) : acc), userAgentString);
```
Value should be defined here so we can skip this I think.

Reply:
Ah, of course; thanks !